packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditor.integration.test.tsx

```diff
@@ -149,14 +168,35 @@ describe('TableColumnSchemaEditor', () => {
 
     const { user, saveButton } = await setUp({
       entityId: mockTableEntityData.id,
+      open: true,
+      onColumnsUpdated: mockOnColumnsUpdated,
+      onCancel: mockOnCancel,
     })
 
     // Add a column
     await addColumnModelToForm('newColumnName', user)
     await user.click(saveButton)
 
-    await waitFor(() => {
-      expect(updateTableSpy).toHaveBeenCalled()
+    await waitFor(async () => {
+      expect(createTableUpdateTransactionRequestSpy).toHaveBeenCalledWith(
+        MOCK_ACCESS_TOKEN,
+        mockTableEntityData.id,
+        mockTableQueryData.columnModels,
+        [
+          ...mockTableQueryData.columnModels!,
+          {
+            name: 'newColumnName',
+            columnType: 'STRING',
+            maximumSize: 50,
+          },
+        ],
+      )
+
+      expect(updateTableSpy).toHaveBeenCalledWith(
+        await createTableUpdateTransactionRequestSpy.mock.results[0].value,
+        MOCK_ACCESS_TOKEN,
+      )
+      expect(mockOnColumnsUpdated).toHaveBeenCalled()
```

Comment:
For each test with a successful update call, verify 3 things
1. We passed the expected data to the update fn (skipping the react-query layer, we just verify a call to `createTableUpdateTransactionRequest`)
2. Verify that `SynapseClient.updateTable` is called with the result of `createTableUpdateTransactionRequest`
3. Verify that our `onColumnUpdated` callback is invoked

#1 is very important that we get right, else we can have bugs that can result in unexpected data loss. For example, we need to capture the nuance between 
* Updating column 123 to now have the schema defined by column 456. Data should be preserved.
* Deleting column 123 and adding column 456 does not preserve data

This PR fixes bugs where the user tries to do an update but the implementation erroneously did a delete + create